cpp/src/arrow/compute/api_scalar.h

```diff
@@ -266,9 +267,10 @@ Result<Datum> Power(const Datum& left, const Datum& right,
 /// \param[in] ctx the function execution context, optional
 /// \return the element-wise maximum
 ARROW_EXPORT
-Result<Datum> ElementWiseMax(const std::vector<Datum>& args,
-                             ElementWiseAggregateOptions options = {},
-                             ExecContext* ctx = NULLPTR);
+Result<Datum> ElementWiseMax(
+    const std::vector<Datum>& args,
+    ElementWiseAggregateOptions options = ElementWiseAggregateOptions::Defaults(),
```

Comment:
This change to use the `::Defaults()` pattern is not necessarily needed (I was copying it from the aggregate kernels, but the scalar kernels don't do this), so I can revert this part.